File: client/src/main/java/it/unibo/pcd/client/Tile.java
package it.unibo.pcd.client;

import java.awt.*;

class Tile implements Comparable<Tile> {
	private Image image;
	private int originalPosition;
	private int currentPosition;

    public Tile(final Image image, final int originalPosition, final int currentPosition) {
        this.image = image;
        this.originalPosition = originalPosition;
        this.currentPosition = currentPosition;
    }
    
    public Image getImage() {
    	return image;
    }
    
    public boolean isInRightPlace() {
    	return currentPosition == originalPosition;
    }
    
    public int getCurrentPosition() {
    	return currentPosition;
    }
    
    public void setCurrentPosition(final int newPosition) {
    	currentPosition = newPosition;
    }

	@Override
	public int compareTo(Tile other) {
		return Integer.compare(this.currentPosition, other.currentPosition);
	}
}

File: client/src/main/kotlin/it/unibo/pcd/client/ui/Tile.kt
package it.unibo.pcd.client.ui

import java.awt.Image
class Tile(val image: Image,
           private val originalPosition: Int,
           var currentPosition: Int) : Comparable<Tile> {

    val isInRightPlace: Boolean
        get() = currentPosition == originalPosition

    override operator fun compareTo(other: Tile): Int {
        return currentPosition.compareTo(other.currentPosition)
    }
}